### PR TITLE
Hystrix Dashboard Cannot display anything for Turbine Stream

### LINKED ISSUE #2475: [Edgware.RELEASE]Hystrix Dashboard Cannot display anything for Turbine Stream
I update my samples to Edgware.RELEASE  then I find that Hystrix Dashboard Cannot display anything for Turbine Stream.

Turbine Stream：<http://localhost:8989> , When I call it, I Can get data like below:

```json
data: {"rollingCountFallbackFailure":0,"rollingCountFallbackSuccess":0,"propertyValue_circuitBreakerRequestVolumeThreshold":"20","propertyValue_circuitBreakerForceOpen":false,"propertyValue_metricsRollingStatisticalWindowInMilliseconds":"10000","latencyTotal_mean":0,"type":"HystrixCommand","rollingCountResponsesFromCache":0,"TypeAndName":"TypeAndName=>HystrixCommand_zuul-server.microservice-provider-user","rollingCountTimeout":0,"propertyValue_executionIsolationStrategy":"SEMAPHORE","instanceId":"zuul-server:-3102666220667402369","rollingCountFailure":0,"rollingCountExceptionsThrown":0,"latencyExecute_mean":0,"isCircuitBreakerOpen":false,"errorCount":0,"group":"RibbonCommand","rollingCountSemaphoreRejected":0,"latencyTotal":{"0":0,"25":0,"50":0,"75":0,"90":0,"95":0,"99":0,"99.5":0,"100":0},"requestCount":0,"rollingCountCollapsedRequests":0,"rollingCountShortCircuited":0,"latencyExecute":{"0":0,"25":0,"50":0,"75":0,"90":0,"95":0,"99":0,"99.5":0,"100":0},"propertyValue_circuitBreakerSleepWindowInMilliseconds":"5000","currentConcurrentExecutionCount":0,"propertyValue_executionIsolationSemaphoreMaxConcurrentRequests":"100","errorPercentage":0,"rollingCountThreadPoolRejected":0,"propertyValue_circuitBreakerEnabled":true,"propertyValue_executionIsolationThreadInterruptOnTimeout":true,"propertyValue_requestCacheEnabled":true,"rollingCountFallbackRejection":0,"propertyValue_requestLogEnabled":true,"rollingCountSuccess":0,"propertyValue_fallbackIsolationSemaphoreMaxConcurrentRequests":"10","InstanceKey":"InstanceKey=>zuul-server:-3102666220667402369","propertyValue_circuitBreakerErrorThresholdPercentage":"50","propertyValue_circuitBreakerForceClosed":false,"name":"zuul-server.microservice-provider-user","reportingHosts":1,"propertyValue_executionIsolationThreadPoolKeyOverride":"null","propertyValue_executionIsolationThreadTimeoutInMilliseconds":"2000"}
```

But When I put this URL into Hystrix Dashboard, I Will never get the graph!
hystrix-dashboard

Then I downgrade my Code to `Dalston.SR4` , everything works! I think it's a Turbine Server bug of `Edgware.RELEASE` .

CODE IS HERE: <https://gitee.com/itmuch/spring-cloud-yes> . Hope you Can help me.

@ryanjbaxter @dsyer @royclarkson

### PR DESCRIPTION
**Pull request by daniellavoie**
Closes #2475

### CODE REVIEW

File: spring-cloud-netflix-turbine-stream/src/test/java/org/springframework/cloud/netflix/turbine/stream/TurbineStreamTests.java [hunk 1]
```diff
@@ -129,21 +128,17 @@ public void contextLoads() throws Exception {
...
-				&& StringUtils.countOccurrencesOf(builder.toString(), "\n") < 2) {
-			read = response.getBody().read(bytes, 0, bytes.length);
-			if (read > 0) {
-				latch.countDown();
-				builder.append(new String(bytes, 0, read));
+
+		String responseBody = "";
+		try (BufferedReader buffer = new BufferedReader(
+				new InputStreamReader(response.getBody()))) {
+			for (int i = 0; i <= 3; i++) {
```
Comment (ryanjbaxter) [inline spring-cloud-netflix-turbine-stream/src/test/java/org/springframework/cloud/netflix/turbine/stream/TurbineStreamTests.java]: Not sure i understand why this for loop is only executed 4 times, could you explain?

Comment (daniellavoie) [reply spring-cloud-netflix-turbine-stream/src/test/java/org/springframework/cloud/netflix/turbine/stream/TurbineStreamTests.java]: 4 calls are required to retreive the entire payload. With 3 calls, we only get

```

event: message
data: {Ping}
```

the 4th call gets you the metrics data line. (discovered that while testing).

Comment (ryanjbaxter) [reply spring-cloud-netflix-turbine-stream/src/test/java/org/springframework/cloud/netflix/turbine/stream/TurbineStreamTests.java]: Could we then just loop through the buffer looking for the specific metrics data line and then break out of the loop?  That way this test is a little less fragile to any future changes.

Comment (daniellavoie) [reply spring-cloud-netflix-turbine-stream/src/test/java/org/springframework/cloud/netflix/turbine/stream/TurbineStreamTests.java]: That's reasonable proposition

**Pull request merged**